### DISCUSSION TITLE
Generating single Query and Mutations on DgsConstants for multiple schemas

### DISCUSSION THREAD

**Discussion by jemieeffendy2** [Netflix/dgs-framework#1275] [Ideas]
Recently, we found that when generating code of multiple schemas, all of the queries and mutations are not combined into a single class, which leads into a compilation issue.

I would like to make a change on the generator to resolve that situation on the codegen. What do you think?

**Comment by srinivasankavitha** [comment]
Can you please provide more details on the error? Codegen assumes each schema is complete but the schema can be define across different files. If they are completely different schemas, then those need to be in different modules with codegen applied individually to each of them. If the schema files belong to the same module, then each file can extend the schema by `extend type Query {...}`

**Comment by GrantGochnauer** [comment]
I was trying to do something similar actually. I have 2 different DGS services which are deployed separately. They each have their own schema. I then have a 3rd project that I'm using as a CLI interface into those 2 services and I'd like to copy the schema files from the first 2 projects into my 3rd CLI project so I can generate all the supporting java types. 

What I tried was this in my gradle file but it didn't work. I had the schema files in 2 different folders and then I was hoping to generate two sets of supporting java files but in their own package (e.g. `platform` vs `iam`)

```
tasks.register('generatePlatformApi')  {
	schemaPaths = ["${projectDir}/src/main/resources/schema/platform"]
	packageName = 'com.xxx.cloud.migration.graphql.platform' // The package name to use to generate sources
	generateClientv2 = true // Enable generating the type safe query API
	typeMapping = [
			"Url"               : "java.net.URL",
			"DateTime"          : "java.time.OffsetDateTime",
			"JSON"              : "java.util.Map",
			"Locale"            : "java.util.Locale",
			"Long"              : "java.lang.Long",
			"DocumentConnection": "graphql.relay.SimpleListConnection<Document>",
			"UUID"              : "java.util.UUID"
	]
}

tasks.register('generateIAMApi') {
	schemaPaths = ["${projectDir}/src/main/resources/schema/iam"]
	packageName = 'com.xxx.cloud.migration.graphql.iam' // The package name to use to generate sources
	generateClientv2 = true // Enable generating the type safe query API
	typeMapping = [
			"Url"               : "java.net.URL",
			"DateTime"          : "java.time.OffsetDateTime",
			"JSON"              : "java.util.Map",
			"Locale"            : "java.util.Locale",
			"Long"              : "java.lang.Long",
			"DocumentConnection": "graphql.relay.SimpleListConnection<Document>",
			"UUID"              : "java.util.UUID"
	]
}

tasks.named('generateJava') {
	dependsOn generatePlatformApi, generateIAMApi
}
```

**Reply by srinivasankavitha** [reply]
If you would like to have 2 separate set of files, it would be better to run codegen task twice by organizing them in different modules in the same project. They would have their own build.gradle in which you can configure the tasks independently. It is not possible to have the same codegen task do this. For each task, all the configured schema paths will be combined together.

**Comment by GrantGochnauer** [comment]
Thanks @srinivasankavitha. I'm going to give this a shot.
